### ISSUE TITLE
executeOnText report different from eslint command line

### ISSUE THREAD

**Issue by noyobo** [eslint/eslint#7826]
**Tell us about your environment**

* **ESLint Version:** 3.12.2
* **Node Version:** v7.3.0
* **npm Version:** 3.10.10

**What parser (default, Babel-ESLint, etc.) are you using?**

- Babel-ESLint

**Please show your full configuration:**

```js
// .eslintrc.js
module.exports = {
  'parser': 'babel-eslint',
  'plugins': [
    'react'
  ],
  'parserOptions': { 
    'sourceType': 'module',
    'ecmaVersion': 6,
    'ecmaFeatures': {
      'jsx': true
    }
  },
  'ecmaFeatures': {
    'arrowFunctions': true,
  },
  'rules': {
    'no-undef': 2
  }
};

```

**What did you do? Please include the actual source code causing the issue.**

js file `foo.js`
```js
import React from 'react';

export default class Foo extends React.Component {

  constructor(props) {
    super(props);
  };

  _handleClick = () => {

  }

  render() {
    return ( 
      <div onClick={this._handleClick} id="hello">
        hello
      </div>
    );
  }
}
```

```bash
eslint -c ./.eslintrc.js foo.js  # There are no errors
```

But use Node api `CLIEngine`

```js

import { CLIEngine } from 'eslint';

const cli = new CLIEngine({
  useEslintrc: false,
  configFile: require.resolve('./.eslintrc.js')
});

cosnt report = cli.executeOnText(/* foo.js content*/, 'foo.js')
```

report.results[0] output 

```
    { filePath: '../foo.js',
      messages:
       [ { ruleId: 'no-undef',
           severity: 2,
           message: '\'_handleClick\' is not defined.',
           line: 9,
           column: 3,
           nodeType: 'Identifier',
           source: '  _handleClick = () => {' } ],
      errorCount: 1,
      warningCount: 0,
      source: 'import React from \'react\';\n\nexport default class Foo extends React.Component {\n\n  constructor(props) {\n    super(props);\n  };\n\n  _handleClick = () => {\n\n  }\n\n  render() {\n    return ( \n      <div onClick={this._handleClick} id="hello">\n        hello\n      </div>\n    );\n  }\n}' }
```

**What did you expect to happen?**

`executeOnText` result equal eslint command line result.

**What actually happened? Please include the actual, raw output from ESLint.**

**Comment by ilyavolodin** [comment]
@noyobo `executeOnText` skips a lot of things (including resolving config file and loading plugins). If you want to use plugins you should either use [`CLIEngine.executeOnFiles`](http://eslint.org/docs/developer-guide/nodejs-api#executeonfiles) or manually load plugins by using [`CLIEngine.addPlugin`](http://eslint.org/docs/developer-guide/nodejs-api#addplugin)
This is behaving as intended.

**Comment by noyobo** [comment]
Thank you
